### DISCUSSION TITLE
About grpc java version support ?

### DISCUSSION THREAD

**Discussion by shalk** [grpc/grpc-java#11630] [Q&A] [answered]
I found gRPC java minor release really monthly. But We can not upgrade the version too often.

Is any release policy or version support  We can follow ?  like https://protobuf.dev/support/version-support/#java

**Comment by ejona86** [comment, marked as answer]
What in particular do you like about that protobuf link? It just shows how they support older major versions for longer when they make a new major version. When a new minor version is released, it marks the end of support for the previous release.  gRPC Java has no planned new major version. 

Releases are every six weeks. The [two most recent releases](https://grpc.io/docs/what-is-grpc/faq/#how-long-are-grpc-releases-supported-for) are supported. Historically, for severe security issues we have fixed even older releases when easily able, but in general we optimize to make upgrading easy (e.g., by not making new major versions) instead of supporting very old versions.

**Reply by shalk** [reply]
Thank you for your reply ! The FAQ is very useful !

In our case, the link show `Public Support Until`, So we will upgrade the version before that time. And We will align to the lts version.

I think i will upgrade gRPC version every 3 month with two most recent releases.

**Reply by ejona86** [reply]
For protobuf, the old major version isn't an LTS of sorts. Yes, it has extended support, but that's to give the ecosystem time to get _off_ of the version. If that extended support is encouraging you to _stay_ on the version, that is the opposite of the intention. And it will cause some problems, as the ecosystem as a whole has to regenerate code with the newer version of protoc, as 5.x won't support 3.x generated code so the ecosystem can't start using a 5.x until it is off 3.x.
